worlds/terraria/__init__.py

```diff
@@ -66,58 +68,84 @@ class TerrariaWorld(World):
     goal_locations: Set[str]
 
     def generate_early(self) -> None:
-        goal, goal_locations = goals[self.options.goal.value]
+        self.options.goal
+        goal_id = self.options.goal.value
+        goal, goal_locations = goals[goal_id]
         ter_goals = {}
         goal_items = set()
         for location in goal_locations:
-            _, flags, _, _ = rules[rule_indices[location]]
+            flags = rules[rule_indices[location]].flags
+            if not self.options.calamity.value and "Calamity" in flags:
+                logging.warning(
+                    f"Terraria goal `{Goal.name_lookup[goal_id]}`, which requires Calamity, was selected with Calamity disabled; enabling Calamity"
+                )
+                self.options.calamity.value = True
+
             item = flags.get("Item") or f"Post-{location}"
             ter_goals[item] = location
             goal_items.add(item)
 
-        achievements = self.options.achievements.value
+        self.calamity = self.options.calamity.value
+        self.getfixedboi = self.options.getfixedboi.value
```

Comment:
Not a huge fan of this type of pattern -- if you have access to the world class, you have access to the option value, so just sticking this on the world class feels unnecessary to me